### DISCUSSION TITLE
Cypress not catching click event , API call is not triggered

### DISCUSSION THREAD

**Discussion by Gayathri-Kesavakannan** [cypress-io/cypress#19529] [Questions and Help]
Cypress not catching click event triggered hence api end points are not called for multiple containers 

It’s very difficult to explain issue , tried my best any hints are really appreciated 

I can’t share any application details 

I have a main UI application interacting with couple of backend components.API calls are made to endpoints with the same base URL but api containers are different based on component 

Manual Test steps; 
1. Visit application url In main UI click search link  
2. New tab/form (in same window) is opened which will interact with component 1 
3. New tab is a form, fill in details and click verify button
4. 2 api calls are triggered 1 : button click event success for container 1  2: api call to container 2
5. On successful 2api call , page returns to main application page with search result details 
In cypress runner for step 4 only 1 api  call for container 1 is triggered with 200 success response 
Unfortunately second api call for container 2 was not triggered 

Asked developers abt this , suggestion was cypress is not catching triggers , container 2 is not reachable from container 1  

I am stuck here , not sure what to search too or ask developers. Any one noticed similar issue ?? Thanks

**Comment by portyu9** [comment]
Hi @Gayathri-Kesavakannan 

The click itself is working—the first API returning `200` proves the click event fired.

The likely problem is that your workflow opens the form in a **new tab/window**. Keep the navigation in Cypress's current tab instead:

```js
cy.get('a[href*="search"]')
  .invoke('removeAttr', 'target')
  .click()
```

Then observe the second request directly:

```js
cy.intercept('POST', '**/container2/**').as('container2')

// perform form action

cy.wait('@container2')
  .its('response.statusCode')
  .should('eq', 200)
```

If `@container2` never appears, the browser did not make that request; inspect the application console/state after the first API response.

If container 1 is supposed to call container 2 **server-to-server**, Cypress cannot trigger or intercept that internal request—the backend/container connectivity must be fixed.

Please mark this answer as accepted if it helped.
